auth/services/contract/notary_test.go

```diff
@@ -0,0 +1,182 @@
+	})
+}
+
+func Test_contractNotaryService_DrawUpContract(t *testing.T) {
+	type testContext struct {
+		ctrl         *gomock.Controller
+		nameResolver *vdr.MockNameResolver
+		notary       contractNotaryService
+	}
+	buildContext := func(t *testing.T) *testContext {
+		ctrl := gomock.NewController(t)
+		ctx := &testContext{
+			ctrl:         ctrl,
+			nameResolver: vdr.NewMockNameResolver(ctrl),
+		}
+		notary := contractNotaryService{
+			nameResolver:     ctx.nameResolver,
+			contractValidity: 15 * time.Minute,
+		}
+		ctx.notary = notary
+		return ctx
+	}
+
+	template := contract.Template{
+		Template: "Organisation Name: {{legal_entity}}, valid from {{valid_from}} to {{valid_to}}",
+	}
+	orgID := *vdr.RandomDID
+	// Add 1 second so !time.Zero()
+	validFrom := time.Time{}.Add(time.Second)
+	duration := 10 * time.Minute
+
+	t.Run("draw up valid contract", func(t *testing.T) {
+		ctx := buildContext(t)
+		defer ctx.ctrl.Finish()
+
+		ctx.nameResolver.EXPECT().Resolve(gomock.Any()).AnyTimes().Return("CareBears", nil)
+
+		drawnUpContract, err := ctx.notary.DrawUpContract(template, orgID, validFrom, duration)
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		assert.NotNil(t, drawnUpContract)
+		assert.Equal(t, "Organisation Name: CareBears, valid from maandag, 1 januari 0001 00:19:33 to maandag, 1 januari 0001 00:29:33", drawnUpContract.RawContractText)
+	})
+
+	t.Run("no given duration uses default", func(t *testing.T) {
+		ctx := buildContext(t)
+		defer ctx.ctrl.Finish()
+
+		ctx.nameResolver.EXPECT().Resolve(gomock.Any()).AnyTimes().Return("CareBears", nil)
+
+		drawnUpContract, err := ctx.notary.DrawUpContract(template, orgID, validFrom, 0)
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		assert.NotNil(t, drawnUpContract)
+		assert.Equal(t, "Organisation Name: CareBears, valid from maandag, 1 januari 0001 00:19:33 to maandag, 1 januari 0001 00:34:33", drawnUpContract.RawContractText)
+	})
+
+	t.Run("no given time uses time.Now()", func(t *testing.T) {
+		ctx := buildContext(t)
+		defer ctx.ctrl.Finish()
+
+		ctx.nameResolver.EXPECT().Resolve(gomock.Any()).AnyTimes().Return("CareBears", nil)
+
+		timenow = func() time.Time {
+			return time.Time{}.Add(10 * time.Second)
+		}
+		defer func() { timenow = time.Now }()
+		drawnUpContract, err := ctx.notary.DrawUpContract(template, orgID, time.Time{}, 0)
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		assert.NotNil(t, drawnUpContract)
+		assert.Equal(t, "Organisation Name: CareBears, valid from maandag, 1 januari 0001 00:19:42 to maandag, 1 januari 0001 00:34:42", drawnUpContract.RawContractText)
+	})
+
+	t.Run("nok - unknown private key", func(t *testing.T) {
+		t.SkipNow()
+		ctx := buildContext(t)
+		defer ctx.ctrl.Finish()
+
+		drawnUpContract, err := ctx.notary.DrawUpContract(template, orgID, validFrom, duration)
+		if assert.Error(t, err) {
+			assert.Equal(t, "could not draw up contract: organization is not managed by this node: missing organization private key", err.Error())
```

Comment:
how can this be if the check has been disabled?

Reply:
fixed